web/src/pages/api/user/confirmEmail.ts

```diff
@@ -1,23 +1,35 @@
 import { User } from "@prisma/client";
 import { NextApiRequest, NextApiResponse } from "next";
 
-import { prisma } from "../../../../server/db/client";
-import { validateMethod } from "../../../../utils/validation";
+import { prisma } from "../../../server/db/client";
+import { validateJWT } from "../../../utils/jwt";
+import { validateMethod } from "../../../utils/validation";
 
 export default async function confirmEmail(
   req: NextApiRequest,
   res: NextApiResponse
 ) {
   if (!validateMethod("GET", req, res)) return;
 
-  const { userID } = req.query;
+  const userID = req.query.id;
+  const token = req.query.token;
 
-  if (Array.isArray(userID) || userID === undefined) {
+  if (
+    Array.isArray(userID) ||
+    userID === undefined ||
+    Array.isArray(token) ||
+    token === undefined
+  ) {

```

Comment:
This can be reduced to 

`if (!userId || !token || Array.isArray(userId) || Array.isArray(token))`